docs/design/coreclr/botr/logging.md

```diff
@@ -0,0 +1,169 @@
+-----------------------
+
+For advanced scenarios, EventSource and the C++ event wrappers both support standard event tracing systems on their respective platforms (ETW on Windows, LTTNG on Linux). EventPipe itself does not integrate with these systems, but the runtime's layers on top of EventPipe will also use them if they are enabled.
+On Windows you can use standard [ETW](https://learn.microsoft.com/en-us/windows-hardware/test/wpt/event-tracing-for-windows) tools like [WPR](https://learn.microsoft.com/en-us/windows-hardware/test/wpt/windows-performance-recorder) with its '.NET Activity' scenario.
+On Linux you can use [LTTNG](https://lttng.org/), and the [perfcollect script](https://learn.microsoft.com/en-us/dotnet/core/diagnostics/trace-perfcollect-lttng) is a useful tool you can use for a more ETW-like experience.
+
+StressLog
+------------
+The StressLog is a circular buffer inside the runtime process that usually does not escape, and most StressLog messages are available in retail builds, which makes it useful for troubleshooting issues with GC or other subsystems in production scenarios. To enable it, you can set the `DOTNET_StressLog` environment variable to `1`, and you can configure it by setting the `DOTNET_LogFacility`, `DOTNET_LogFacility2` and `DOTNET_LogLevel` environment variables.
+
+The `DOTNET_StressLogSize` environment variable configures how large the per-thread buffer for log messages is, and `DOTNET_TotalStressLogSize` allows setting a process-wide cap to avoid exhausting memory when you have many threads. Raising the size will allow you to capture more information in scenarios where the buffer fills too quickly and you're searching for a needle in a haystack. These variables are both specified in bytes. For example to set the per-thread buffer size to 20MB, you would set `DOTNET_StressLogSize=20971520`, and then you could set a process-wide limit of 400MB with `DOTNET_TotalStressLogSize=419430400` if you know you're unlikely to have more than 20 threads.
+
+If you are in a situation where you can't attach a debugger to the process or get a crash dump containing the buffer, the `DOTNET_StressLogFilename` environment variable can be used to write the StressLog's contents to a file instead.
+
+To write your own messages to the StressLog from C++, you can use the `STRESS_LOG_N(facility, level, msg, ...)` macros, i.e. `STRESS_LOG1(LF_GC, LL_ERROR, "A significant but non-fatal error occurred in the garbage collector! Here's my favorite number: %d\n", 42)` where the first argument is one or more logging facilities (you can combine them using `|` i.e. `LF_GC | LF_GCROOTS`), the second argument is a severity level, and the third argument is the log message format string. See [log facilities and levels](#log-facilities-and-levels), below, for more information.
+
+You can't write your own messages to the StressLog from C#. If for some reason you need to while testing, perhaps you could expose it via a custom icall.
```

Comment:
icall is a Mono term. Calls from the managed environment to the native coreclr codebase are called fcalls or qcalls. (And we are trying to get rid of all significant FCalls in favor of having only QCalls. It's a slow, ongoing project.)